test/linalg.jl

```diff
@@ -62,3 +64,57 @@ end
         end
     end
 end
+
+###### Test jacobian matrix ######
+params(A::SparseMatrixCSC) = params(A.nzval)
```

Comment:
Here, `params` is just a regular function name. If one wants to use a function defined in SparseArrays, he needs to implement `SparseArrays.params(...) = ...`.

Reply:
I get it. However, I feel confused about the params(A.nzval), where is the definition of params here? Type of A.nzval is vector, so may be params would be implemented before the definition of params on A? I noticed params have been implemented on dense arrays below.

Reply:
This is the definition.

Reply:
These are just helper functions to define what parameters we're calculating gradients with respect to, i.e, what `W` is in `∂L/∂W`. For test purposes, we simply take the matrix value as the gradient, which is why here we define `params(X) = X` for the dense array.

Recall our previous meeting, I believe @GiggleLiu's idea is to only treat the non-zero part of the sparse array as the parameters, i.e., `params(X::SparseMatrixCSC) = A.nzval`. (Maybe we should leave a comment here to indicate future readers)

These are just helpers, I believe the name is deliberately chosen to mimic other packages; they are new functions instead of methods of existing functions and they don't  really interact with other packages.